fairlearn/metrics/_multi_curve_plot.py

```diff
@@ -0,0 +1,126 @@
+from ..utils._input_validation import _validate_and_reformat_labels, _validate_and_reformat_labels_and_sf, check_consistent_length
+from ..postprocessing._plotting import _MATPLOTLIB_IMPORT_ERROR_MESSAGE
+from ._make_derived_metric import _DerivedMetric
+from typing import Callable, List, Union
+
+def plot_model_comparison(
+    *,
+    x_axis_metric: Callable[..., Union[float, int]],
+    y_axis_metric: Callable[..., Union[float, int]],
+    y_true,
+    y_preds,
+    sensitive_features,
+    show_plot: bool,
+    ax=None,
+    **kwargs
+    ):
+    """Plot a model comparison
+
+    Parameters
+    ----------
+    x_axis_metric : Callable
+        The (aggregating) metric function for the x-axis
+        The passed metric function must take `y_true, y_pred`, and possibly `sensitive_values`.
+        If the metric is grouped, it must aggregate results. For instance, use
+        `make_derived_metric(metric=balanced_accuracy_score, transform='group_min')`
+        to aggregate the `balanced_accuracy_score`.
+
+    y_axis_metric : Callable
+        The (aggregating) metric function for the y-axis, similar to x_axis_metric.
+        The passed metric function must take `y_true, y_pred`, and possibly `sensitive_values`.
+        If the metric is grouped, it must aggregate results.
+
+    y_true : List, pandas.Series, numpy.ndarray, pandas.DataFrame
+        The ground-truth labels (for classification) or target values (for regression).
+
+    y_pred : List, pandas.Series, numpy.ndarray, pandas.DataFrame
+        The predictions.
+
+    sensitive_features : List, pandas.Series, dict of 1d arrays, numpy.ndarray, pandas.DataFrame, optional
+        The sensitive features which should be used to create the subgroups.
+        At least one sensitive feature must be provided.
+        All names (whether on pandas objects or dictionary keys) must be strings.
+        We also forbid DataFrames with column names of ``None``.
+        For cases where no names are provided we generate names ``sensitive_feature_[n]``.
+    
+    show_plot : bool, optional
```

Comment:
I am responsible for introducing this in other plotting functions but I believe we don't want this anymore. Instead, users can do plt.show() afterwards.